experimental/packages/opentelemetry-sdk-metrics-base/src/aggregator/Sum.ts

```diff
@@ -37,29 +44,43 @@ export class SumAccumulation implements Accumulation {
 export class SumAggregator implements Aggregator<SumAccumulation> {
   public kind: AggregatorKind.SUM = AggregatorKind.SUM;
 
-  createAccumulation() {
-    return new SumAccumulation();
+  constructor (public monotonic: boolean) {}
+
+  createAccumulation(startTime: HrTime) {
+    return new SumAccumulation(startTime, this.monotonic);
   }
 
   /**
    * Returns the result of the merge of the given accumulations.
    */
   merge(previous: SumAccumulation, delta: SumAccumulation): SumAccumulation {
-    return new SumAccumulation(previous.toPointValue() + delta.toPointValue());
+    if (delta.reset) {
+      return new SumAccumulation(delta.startTime, this.monotonic, delta.toPointValue(), delta.reset);
+    }
+    return new SumAccumulation(previous.startTime, this.monotonic, previous.toPointValue() + delta.toPointValue());
   }
 
   /**
    * Returns a new DELTA aggregation by comparing two cumulative measurements.
    */
   diff(previous: SumAccumulation, current: SumAccumulation): SumAccumulation {
-    return new SumAccumulation(current.toPointValue() - previous.toPointValue());
+    const prevPv = previous.toPointValue();
+    const currPv = current.toPointValue();
+    /**
+     * If the SumAggregator is a monotonic one and the previous point value is
+     * greater than the current one, a reset is deemed to be happened.
+     * Return the current point value to prevent the value from been reset.
+     */
+    if (this.monotonic && (prevPv > currPv)) {
+      return new SumAccumulation(current.startTime, this.monotonic, currPv, true);
+    }
+    return new SumAccumulation(current.startTime, this.monotonic, current.toPointValue() - previous.toPointValue());
```

Comment:
Minification is helped if we use the existing var and it should be (very slightly) faster

```suggestion
    return new SumAccumulation(current.startTime, this.monotonic, currPv - prevPv);
```